lib/OpenQA/Worker/CommandHandler.pm

```diff
@@ -0,0 +1,185 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::CommandHandler;
+use Mojo::Base 'Mojo::EventEmitter';
+
+use OpenQA::Utils qw(log_error log_debug log_warning log_info);
+
+use POSIX ':sys_wait_h';
+use Data::Dump 'pp';
+
+has 'client';
+
+sub new {
+    my ($class, $client) = @_;
+
+    return $class->SUPER::new(client => $client);
+}
+
+sub handle_command {
+    my ($self, $tx, $json) = @_;
+
+    my $client             = $self->client;
+    my $worker             = $client->worker;
+    my $current_job        = $worker->current_job;
+    my $webui_host         = $client->webui_host;
+    my $current_webui_host = $worker->current_webui_host // 'unknown web UI host';
+
+    # handle responses to our own messages which are indicated by 'result'
+    if ($json->{result}) {
+        # handle response to update_status and filter known images
+        if (my $known_images = $json->{known_images}) {
+            if (!$current_job) {
+                log_warning("Ignoring 'known_images' from $webui_host because no job is running.");
+                return undef;
+            }
+            # FIXME: Also before refactoring upload_images() did not take any arguments. This merely
+            #        uploads images but does not filter known images as the comment in the previous code
+            #        stated. Is this wanted?
```

Comment:
Don't add disabled code as nobody but you will understand why this is not active. Better leave it out and keep in your local git repository, either in `git stash` or a temporary "WIP"-commit or ticket

Reply:
Ok. Since this might be even related to the flaky test I'm investigating right now I'll have another look at this code. If it really serves no purpose I'm getting rid of it in a separate commit (we can be reverted individually if needed).